src/ui/database/report/ReportDatabasePageContainer.js

```diff
@@ -16,4 +18,7 @@ export const mapDispatchToProps = (dispatch, { match: { params } }) => ({
   },
 });
 
-export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ReportDatabasePage));
+export default withRouter(connect(
+  mapStateToProps,
+  mapDispatchToProps,
+)(withPermissions(ROLE_SUPERUSER)(ReportDatabasePage)));
```

Comment:
avoid too much nesting. pls code this like what you did on other container components